src/extra/Markdown/index.tsx

```diff
@@ -138,11 +139,110 @@ type components = {
 	[element: string]: (props: any) => JSX.Element | React.ReactElement | Element;
 };
 
+interface Decorator {
+	match: RegExp;
+	component: string;
+}
+
+interface DecoratorPluginOptions {
+	decorators: Decorator[];
+}
+
+const decoratorPlugin = (options: DecoratorPluginOptions) => {
+	options = options || { decorators: [] };
+
+	if (!options.decorators || !options.decorators.length) {
+		return;
+	}
+
+	const targets: Array<{ decorator: Decorator; node: any }> = [];
+
+	const visitor = (node: any) => {
+		const matchedDecorator = options.decorators.find((decorator) => {
+			return (
+				node.children &&
```

Comment:
Use optional operand, so this will be `node.children?.[0]?.type === 'text' && node.children[0].value`